.ci/release/Jenkinsfile

```
@@ -0,0 +1,229 @@
+
+            Click 'Proceed' to confirm that this step has been completed or Abort to stop the build in order to complete this step.
+            """
+            )
+          }
+        }
+        stage('Review project version') {
+          steps {
+            script {
+              def ver = mvnVersion(showQualifiers: true)
+              def should_continue = input(message: "Current version is ${ver}", parameters: [
+                [
+                  $class: 'ChoiceParameterDefinition',
+                  name: "Do you wish to update the version?",
+                  "choices": ["Yes", "No"],
+                  description: "Selecting 'Yes' will allow you to select the new version in the next step."
+                ]
+              ])
+              if (should_continue == 'Yes'){
+                def new_version = input(message: "Please enter version to change to:", parameters:
+                  [
+                    [
+                      $class: 'StringParameterDefinition',
+                      defaultValue: "${ver}",
+                      description: 'We will run mvn versions:set -DnewVersion=<NEW_VERSION>', name: 'New Version'
+                    ]
+                  ]
+                )
+                sh(name: "mavenVersionUpdate", script: "mvn versions:set -DnewVersion=${new_version}")
+              } else {
+                echo "Skipping version update"
+              }
+            }
+          }
+        }
+        stage('Nexus release') {
+          steps {
+            script {
+              def spid = getVault('nexus')["staging-profile-id"]
+              dir("${BASE_DIR}"){
+                withEnvMask(vars: [[var: "SPID", password: spid]]){
+                  def foundStagingId = nexusFindStagingId(stagingProfileId: "${SPID}", groupId: "co.elastic.apm")
+                  nexusCloseStagingRepository(stagingProfileId: "${SPID}", stagingId: foundStagingId)
+                  nexusReleaseStagingRepository(stagingProfileId: "${SPID}", stagingId: foundStagingId)
+                }
+              }
+            }
+          }
+        }
+        stage('Branch creation') {
+          steps {
+            dir("${BASE_DIR}") {
+              setupAPMGitEmail()
+              script {
+                env.BRANCH_NAME = "tags/${env.TAG_BARE}"
+              }
+              withGitRelease() {
+                script {
+                  def curVer = mvnVersion()
+                  sh(script: "./scripts/jenkins/branch_creation.sh ${curVer}")
+                  def isMajor = sh(script: "./scripts/jenkins/is_major.sh ${curVer}", returnStatus: true)
+                  if (isMajor == 0) {
+                    input message: "This was a major version release. Please update the conf.yml in the docs repo before continuing", ok "Continue"
+                  }
+                }
+                gitPush()
+              } 
```

IIRC, `BRANCH_NAME` is a read only env variable, `withEnv` could help to change the value for the release context, what do you think?

```suggestion
              withEnv(["BRANCH_NAME=tags/${env.TAG_BARE}"]){
                withGitRelease() {
                  script {
                    def curVer = mvnVersion()
                    sh(script: "./scripts/jenkins/branch_creation.sh ${curVer}")
                    def isMajor = sh(script: "./scripts/jenkins/is_major.sh ${curVer}", returnStatus: true)
                    if (isMajor == 0) {
                      input message: "This was a major version release. Please update the conf.yml in the docs repo before continuing", ok "Continue"
                    }
                  }
                  gitPush()
                }
              }
              
```

---

Interesting. I had no problems in setting it directly when I was testing it locally but I agree that if we can use a context-based approach that it is much cleaner. Thanks for pointing this out. I wasn't familiar with `withEnv`.